addons/purchase_stock/models/purchase.py

```diff
@@ -126,7 +126,7 @@ def button_cancel(self):
     def action_view_picking(self):
         """ This function returns an action that display existing picking orders of given purchase order ids. When only one found, show the picking immediately.
         """
-        action = self.env.ref('stock.action_picking_tree_all')
+        action = self.env.ref('stock.action_picking_tree_all').sudo()
```

Comment:
`sudo()` works but is a little bit too overkill and not so robust against vulnerabilities. Instead, it's safer to get the action values with `_from_xml_id`.
Here is an example : 
https://github.com/odoo/odoo/blob/d3bad3316bcde90ff182decb33c9729c54f9f601/addons/stock/models/product.py#L417-L420